### ISSUE TITLE
[a11y]: Tree view component does not expand/collapse node on Enter key

### ISSUE THREAD

**Issue by anagha2891** [carbon-design-system/carbon#18837]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v11

### React version

v17

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker Version 4.0.0

### Description

The expand/collapse feature of the nodes which works on click does not seem to work with Enter key. Its throwing an error instead. 
This issue can be reproduced in the [Carbon Components doc](https://react.carbondesignsystem.com/?path=/docs/components-treeview--overview) if you click Enter a couple of times on a node with expand option

### WCAG 2.1 Violation

Yes

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-treeview--overview

### Steps to reproduce

1. navigate with Tab key to any of the nodes with expand/collapse option for Tree view
2. click on Enter. Nothing happens
3. Click on Enter a couple of times more. 
4. Page errors out.

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by riddhybansal** [comment]
Here is the [slack](https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1742191901681899) conversation for the same .
